test/integration/ocsp_test.go

```diff
@@ -0,0 +1,192 @@
+// +build integration
+
+package integration
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/eggsampler/acme/v2"
+	ocsp_helper "github.com/letsencrypt/boulder/test/ocsp/helper"
+)
+
+func random_domain() string {
+	var bytes [3]byte
+	rand.Read(bytes[:])
+	return hex.EncodeToString(bytes[:]) + ".com"
+}
+
+type client struct {
+	acme.Account
+	acme.Client
+}
+
+func makeClient() (*client, error) {
+	c, err := acme.NewClient(os.Getenv("DIRECTORY"))
+	if err != nil {
+		return nil, fmt.Errorf("Error connecting to acme directory: %v", err)
+	}
+	privKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		return nil, fmt.Errorf("error creating private key: %v", err)
+	}
+	account, err := c.NewAccount(privKey, false, true, "mailto:example@letsencrypt.org")
+	if err != nil {
+		return nil, fmt.Errorf("error creating new account: %v", err)
+	}
+	return &client{account, c}, nil
+}
+
+func addHTTP01Response(token, keyAuthorization string) error {
```

Comment:
To avoid problems down the road I think it would make sense to add a `removeHTTP01Response` that undoes the challtestsrv work in a `defer`.